dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LoginPropertyTest.java

```diff
@@ -0,0 +1,329 @@
+     * LoginProperties: none
+     *
+     * Expected Results:
+     *
+     * User can login with CN (userFilter), but not with UID or SN.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testNoLoginPropertyWithFilter() throws Exception {
+        updateLibertyServer(true);
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userCn, "password"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userUid, "password"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userSn, "password"));
+    }
+
+    /**
+     * Configuration:
+     *
+     * UserFilter: (&(cn=%v)(objectclass=inetorgperson))
+     * LoginProperties: SN
+     *
+     * Expected Results:
+     *
+     * User can login with SN (loginProperty), but not with CN (userFilter) or UID.
+     * When logging in with SN, a warning should be issued that indicates the dynamically
+     * generated filter is used instead of the userFilter.
+     *
+     * Additionally, we expect the user display name to be SN because loginProperty
+     * takes precedence.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testLoginPropertyWithFilter() throws Exception {
+        updateLibertyServer(true, "sn");
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userSn, "password"));
+        assertNotNull("Should find CWIML4505W message.", libertyServer.waitForStringInLog("CWIML4505W"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userCn, "password"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userUid, "password"));
+        assertEquals(userSn, servlet.getUserDisplayName(USER_DN));
+    }
+
+    /**
+     * Configuration:
+     *
+     * UserFilter: not explicitly configured
+     * LoginProperties: SN
+     *
+     * Expected Results:
+     *
+     * User can login with SN, but not with CN or UID.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testLoginProperty() throws Exception {
+        updateLibertyServer(false, "sn");
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userSn, "password"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userUid, "password"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userCn, "password"));
+    }
+
+    /**
+     * Configuration:
+     *
+     * UserFilter: not explicitly configured
+     * LoginProperties: CN, SN
+     *
+     * Expected Results:
+     *
+     * User can login with CN and SN, but not with UID.
+     *
+     * Additionally, we expect the user display name to be cn as this is
```

Comment:
I assume we make assertion on display name because it is mapped to principalName in the input / output mappings for display name? I am not sure if that is the case or not. This would apply to all tests.

If this is the case, explain that where we make this statement.